src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -218,28 +218,23 @@ internal void Append(JsonTokenType tokenType, int startLocation, int length)
 
                 if (Length >= _data.Length - DbRow.Size)
                 {
-                    Enlarge();
+                    Debug.Assert(!_isLocked, "Appending to a locked database");
+
+                    byte[] toReturn = _data;
+                    _data = ArrayPool<byte>.Shared.Rent(toReturn.Length * 2);
+                    Buffer.BlockCopy(toReturn, 0, _data, 0, toReturn.Length);
+
+                    // The data in this rented buffer only conveys the positions and
+                    // lengths of tokens in a document, but no content; so it does not
+                    // need to be cleared.
+                    ArrayPool<byte>.Shared.Return(toReturn);
                 }
 
                 DbRow row = new DbRow(tokenType, startLocation, length);
                 MemoryMarshal.Write(_data.AsSpan(Length), ref row);
                 Length += DbRow.Size;
             }
 
-            private void Enlarge()
```

Comment:
Does moving this in hurt the inlining of the `Append` method since that method has a larger codesize now that the rare occurrence is inlined? Was it previously getting inlined by the JIT. If so, that might hurt perf, so we could revert this change and leave things as is.

Reply:
Per the related comment, I don't think there is ever an automatic inlining when there is a condition, but I added back Enlarge() anyway.